### DISCUSSION TITLE
Query is failing "MATCH (i)-[e]->(o) where id(i) == 'ABC' or id(o) == 'ABC' return e;"

### DISCUSSION THREAD

**Discussion by xs2tarunkukreja** [vesoft-inc/nebula#5582] [Q&A] [answered]
I am trying to find all relationship with an vertex (either in or out).

When I run "MATCH (i)-[e]->(o) where id(i) == 'ABC' return e;" or "MATCH (i)-[e]->(o) where id(o) == 'ABC' return e;" - Both work fine.

But when I run "MATCH (i)-[e]->(o) where id(i) == 'ABC' or id(o) == 'ABC' return e;" - It throw an error  "Scan vertices or edges need to specify a limit number, or limit number can not push down.".

Is there any way to find all the relationship in one query.

**Comment by wey-gu** [comment, marked as answer]
```cypher
MATCH (i)-[e]-() WHERE id(i) == 'ABC' RETURN e;
```

**Reply by wey-gu** [reply]
You should hit an optimizer bug though :)

**Reply by xs2tarunkukreja** [reply]
Thanks.. I will check your suggestion. But I am not sure.. why my query is not working? But the partial queries are working successfully.

Would you please guide me to know the reason of error?

**Reply by xs2tarunkukreja** [reply]
Yeah, it appear as your solution is working.. Please guide me to know why my query is not working.

**Reply by wey-gu** [reply]
It's a unnecessary filter condition thus not yet verified, could you please help report an issue for that, you can share the execution plan in the issue, too.

Just query with prefix: explain like: explain match...

**Reply by xs2tarunkukreja** [reply]
Ok.. Should I report the issue?

**Reply by wey-gu** [reply]
I reported here https://github.com/vesoft-inc/nebula/issues/5583

The issue here is that, instead of do "We know start node" search, this query was planned to scan(index or data) and filter vid from all, which is not optimal, and before 3.5.0, full scan of data without LIMIT sampling is not allowed(but it's changed in 3.5.0), thus you hit this error, although you wouldn't hit error in 3.5.0, it's not recommended to do this query as it's too expansive.

My suggested query expressed the equivalent but will be optimized to explicitly seek the starting node :) as `--` refers to `-->` or `<--`.
